### PR TITLE
updated installation script

### PR DESCRIPTION
**Pull request by abuts**
The  simplified installation script to use on new Horace

### CODE REVIEW

**Review by cmarooney-stfc — COMMENTED** [review]
No individual comments, but it would be good to have recorded that this has been tested ok and what platforms that is on.

Also, is this updated to match the merge, or to match problems found trying to use the merge?

File: admin/horace_install.m [hunk 1]
```diff
@@ -1,21 +1,15 @@
-function [init_folder,her_init_dir,hor_init_dir,use_old_init_path] = ...
+function [init_folder,hor_init_dir,use_old_init_path] = ...
     horace_install(varargin)
 % Install an initialize Horace at the location, where the Horace package
```
[the remark this answers is not in the document]
Comment (abuts) [reply admin/horace_install.m:3]: ?

[the remark this answers is not in the document]
Comment (cmarooney-stfc) [reply admin/horace_install.m:3]: He's saying "install AND initialise" AN->AND comment on line 3, not the change

File: herbert_core/classes/MPIFramework/@MPI_clusters_factory/private/select_and_set_working_parallel_cluster_.m [hunk 2]
```diff
@@ -47,23 +47,24 @@ function select_and_set_working_parallel_cluster_(obj,val)
...
+    obj.framework_available_   = false;
     if strcmp(ME.identifier,'HERBERT:ClusterWrapper:not_available')
-        obj.framework_available_ = false;
         return
     else
         rethrow(ME);
     end
 end
-obj.framework_available_ = true;
+
```
Comment (oerc0122) [inline herbert_core/classes/MPIFramework/@MPI_clusters_factory/private/select_and_set_working_parallel_cluster_.m:69]: Is this form common enough we want to make it a function?

File: horace_core/configuration/@hor_config/hor_config.m [hunk 3]
```diff
@@ -221,7 +221,8 @@
 
         function this = set.pixel_page_size(this, val)
             PixelData.validate_mem_alloc(val);
-            config_store.instance().store_config(this, 'pixel_page_size', val);
+            m_chunk_size = floor(val/36);
```
Comment (oerc0122) [inline horace_core/configuration/@hor_config/hor_config.m:224]: Why `36`? Comment

File: horace_core/sqw/@rundatah/build_bounding_obj.m [hunk 4]
```diff
@@ -80,6 +84,6 @@
 
 if en_min == 0 || en_max == 0 || sign(en_min)*sign(en_max)>0
     en = [en_min*(1-sign(en_min)*eps),en_max*(1+sign(en_max)*eps)];
-else %we want even number of equally spaced bins, to produce 3 odd bin cenres    
+else %we want even number of equally spaced bins, to produce 3 odd bin cenres
```
Comment (oerc0122) [inline horace_core/sqw/@rundatah/build_bounding_obj.m:87]: centres

**Review by oerc0122 — CHANGES_REQUESTED** [review]
You seem to have captured all the mex-files in this? Do we want those as part of the repo or should they be built for each system (which is likely the only way they'll run)

File: _test/test_sqw_file/test_hdf_pix_group.m [hunk 5]
```diff
@@ -15,7 +15,7 @@
             end
             obj = obj@TestCase(class_name);
             ver = matlab_version_num();
-            if ver>9.8
+            if ver>9.08
                 obj.skip_tests = true;                
             else
                 obj.skip_tests = false;
```
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_sqw_file/test_hdf_pix_group.m:21]: its opposite to that. higher versions do not currently work as they changed hdf version and there is ticket to fix that. As it was there in this form, I left this form.

File: CMakeLists.txt [hunk 6]
```diff
@@ -26,7 +29,19 @@ list(APPEND CMAKE_MODULE_PATH "${Horace_ROOT}/cmake"
...
 # This sets the destination for mex build artefacts (used in PACE_AddMex)
-set(Horace_DLL_DIRECTORY "${Horace_ROOT}/horace_core/DLL")
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+   # Mac OS X specific code
+   SET(Matlab_ARCH "_MACI64")
+   option(USE_HORACE_MPI "Use MPI libraries provided with Horace" OFF)   
+ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+   SET(Matlab_ARCH "_GLNXA64")
+   option(USE_HORACE_MPI "Use MPI libraries provided with Horace" ON)
+ELSE()
```
[the remark this answers is not in the document]
Comment (abuts) [reply CMakeLists.txt:39]: ok, this worth doing.

File: _test/test_sqw_file/test_hdf_pix_group.m (hunk 5 above)
[the remark this answers is not in the document]
Comment (oerc0122) [reply _test/test_sqw_file/test_hdf_pix_group.m:21]: Then `obj.skip_tests = ~verLessThan('matlab', '9.8')`

File: CMakeLists.txt [hunk 7]
```diff
@@ -5,6 +5,9 @@ file(READ "${Horace_ROOT}/VERSION" _version)
 string(STRIP "${_version}" _version)
 project("Horace" VERSION "${_version}")
 
+#
+cmake_host_system_information(RESULT CMAKE_HOST_SYSTEM_NAME QUERY OS_NAME)
```
[the remark this answers is not in the document]
Comment (abuts) [reply CMakeLists.txt:9]: this script generates different `OS_NAME` from `if(Win32)` and similar. Why not to use it and may be expand in the future.

File: CMakeLists.txt [hunk 8]
```diff
@@ -26,7 +29,19 @@ list(APPEND CMAKE_MODULE_PATH "${Horace_ROOT}/cmake"
      "${Horace_ROOT}/cmake/external" "${Horace_ROOT}/cmake/shared")
 
 # This sets the destination for mex build artefacts (used in PACE_AddMex)
-set(Horace_DLL_DIRECTORY "${Horace_ROOT}/horace_core/DLL")
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```
[the remark this answers is not in the document]
Comment (abuts) [reply CMakeLists.txt:32]: do not know what you mean here, but neither `${CMAKE_SYSTEM_NAME} EQUAL "Darwin"` nor `CMAKE_SYSTEM_NAME EQUAL "Darwin"` work

File: herbert_core/classes/MPIFramework/@MPI_clusters_factory/private/select_and_set_working_parallel_cluster_.m (hunk 2 above)
Comment (abuts) [reply herbert_core/classes/MPIFramework/@MPI_clusters_factory/private/select_and_set_working_parallel_cluster_.m:69]: do not know -- I do not remember it wider usage.

File: horace_core/configuration/@hor_config/hor_config.m (hunk 3 above)
Comment (oerc0122) [reply horace_core/configuration/@hor_config/hor_config.m:224]: That may be so, but until it is, a comment is needed to explain where `36` comes from.

**Pull request merged**